docs/architecture/architecture.md

```diff
@@ -609,6 +614,133 @@ rootfs by calling `mount(8)` within the counter.  If the devicemapper block devi
+[`gRPC`](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/cri/v1alpha1/runtime/api.proto) 
+[Container Runtime Interface (CRI)](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/container-runtime-interface-v1.md).
+In other words, `kubelet` is a `CRI` client and expects a `CRI` implementation to
+handle the server side of the interface.
+[CRI-O\*](https://github.com/kubernetes-incubator/cri-o) is a `CRI` implementation
+that relies on `OCI` compatible runtimes for managing containers instances.
+
+Clear Containers, being an OCI compatible runtime, naturally fits into the `CRI-O`
+architecture and requirements, and is one of the 2 officially supported `CRI-O` runtimes.
+However, due to the fact that Kubernetes execution units are sets of containers (also
+known as pods) rather than single containers, the Clear Containers runtime needs to
+get extra information to seamlessly integrate with Kubernetes.
+
+### Problem statement
+
+The [Kubernetes\*](https://github.com/kubernetes/kubernetes/) execution unit is a `Pod`,
+i.e. a set of containers sharing the same resources and networking, mount and PID
+namespaces. A pod specification is a set of containeris with an associated list of
+constraints (namespaces, cgroups, hardware resources, security contexts, etc...).
+By default the Clear Containers runtime associates each container with a dedicated
+virtual machine, but that does not work well with pods as letting a set of virtual
+machiness share the same namespaces and resources can be very complex to implement.
+For example allowing two virtual machines to share the same networking or PID namespace
+can be very challenging.
+
+By default the Clear Containers runtime creates a virtual machine for each container,
+but with Kubernetes it will create and launch one virtual machine per pod. Then each
+container within a pod is simply a container inside the virtual machine boundary.
+
+The challenge with Clear Containers when working as a `CRI-O` runtime is to know when
+to create a full virtual machine (for pods) and when to create a new container inside
+a previously created virtual machine. In both cases it will get called with very similar
+arguments, so it needs the help of `CRI-O` to be able to distinguish between a pod
+creation from a container one.
+
+### OCI annotations
+
+In order for the Clear Containers runtime (or any virtual machine  based OCI compatible
+runtime) to be able to understand if it needs to create a full virtual machine or if it
+has to create a new container inside an existing pod's virtual machine, `CRI-O` adds
+specific annotations to the OCI configuration file (`config.json`) which is passed to
+the OCI compatible runtime.
+
+Before calling its runtime, `CRI-O` will always add a `io.kubernetes.cri-o.ContainerType`
+annotation to the `config.json` `OCI` configuration file it produces from the kubelet `CRI`
```

Comment:
I wonder if we can drop the "`OCI` configuration file" text as that has been explained in the previous sentence, so it would become just, "to the `config.json` file it produces..."